custom_components/ppc_smgw/const.py

```diff
@@ -35,6 +39,24 @@
         device_class=SensorDeviceClass.ENERGY,
         state_class=SensorStateClass.TOTAL_INCREASING,
     ),
+    SensorEntityDescription(
+        key="1-1:1.7.0",
```

Comment:
This is the first time I am seeing channel information in OBIS codes (`1-1` means electricity values on channel 1).
Do you have a multi-connection meter or the SMGW connected to multiple meters?

Reply:
This actually makes me wonder how common that would be. Having support for multiple meters/channels would make the detection more complicated and would require the integration to actually understand the OBIS codes